src/url.js

```diff
@@ -156,68 +289,38 @@ function finalizeResourceType(resourceType = "image", type = "upload", urlSuffix
  * @param {type} [options.type='upload'] - The asset's storage type.
  *  For details on all fetch types, see
  * <a href="https://cloudinary.com/documentation/image_transformations#fetching_images_from_remote_locations"
- *  target="_blank">Fetch types</a>. 
- * @param {Object} [options.resource_type='image'] - The type of asset. <p>Possible values:<br/> 
+ *  target="_blank">Fetch types</a>.
+ * @param {Object} [options.resource_type='image'] - The type of asset. <p>Possible values:<br/>
  *  - `image`<br/>
  *  - `video`<br/>
- *  - `raw` 
+ *  - `raw`
  * @return {string} The media asset URL.
  * @see <a href="https://cloudinary.com/documentation/image_transformation_reference" target="_blank">
  *  Available image transformations</a>
  * @see <a href="https://cloudinary.com/documentation/video_transformation_reference" target="_blank">
- *  Available video transformations</a>   
+ *  Available video transformations</a>
  */
 export default function url(publicId, options = {}, config = {}) {
-  var error, prefix, ref, resourceTypeAndType, transformation, transformationString, url, version;
   if (!publicId) {
     return publicId;
   }
+
   if (options instanceof Transformation) {
     options = options.toOptions();
   }
+
   options = defaults({}, options, config, DEFAULT_IMAGE_PARAMS);
+
   if (options.type === 'fetch') {
     options.fetch_format = options.fetch_format || options.format;
     publicId = absolutize(publicId);
   }
```

Comment:
@maoznir , would extract that to a "prepareOptions" or "handleOptions" method

Reply:
done

Reply:
@maoznir Would break the handling of options and publicId (even if we ask "if type==fetch" twice)

Reply:
done